packages/openapi-v3/src/__tests__/unit/enhancers/consolidate-schema.enhancer.unit.ts

```diff
@@ -0,0 +1,197 @@
+        schema: {
+          title: 'loopback.example',
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    };
+    const expectedSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          $ref: '#/components/schemas/loopback.example',
+        },
+      },
+      components: {
+        schemas: {
+          'loopback.example': {
+            title: 'loopback.example',
+            properties: {
+              test: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    };
+    expect(consolidationEnhancer.modifySpec(inputSpec)).to.eql(expectedSpec);
+  });
+
+  it('ignores schema without title property', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    };
+    expect(consolidationEnhancer.modifySpec(inputSpec)).to.eql(inputSpec);
+  });
+
+  it('Avoids naming collision', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          title: 'loopback.example',
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        },
+      },
+      components: {
+        schemas: {
+          'loopback.example': {
+            title: 'Different loopback.example exists',
+            properties: {
+              testDiff: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    };
+    const expectedSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          $ref: '#/components/schemas/loopback.example1',
+        },
+      },
+      components: {
+        schemas: {
+          'loopback.example': {
+            title: 'Different loopback.example exists',
+            properties: {
+              testDiff: {
+                type: 'string',
+              },
+            },
+          },
+          'loopback.example1': {
+            title: 'loopback.example',
+            properties: {
+              test: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    };
+    expect(consolidationEnhancer.modifySpec(inputSpec)).to.eql(expectedSpec);
+  });
+
+  it('If array items has no title, copy parent title if exists', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          myarray: {
```

Comment:
IIUC, you are placing the schema directly under the `paths` object. I don't think that's a valid OpenAPI spec:

- According to https://github.com/OAI/OpenAPI-Specification/blob/3.0.1/versions/3.0.1.md#openapi-object, the top-level object contains `paths` property with `Paths Object` value.
- Paths Object is defined here: https://github.com/OAI/OpenAPI-Specification/blob/3.0.1/versions/3.0.1.md#pathsObject, it is supposed to contain path entries
- Path Item Object can contain HTTP verbs and shared metadata like parameters

Can you please update all tests to use a valid OpenAPI spec document? For example, you can describe a dummy `GET /` endpoint and put the schema to `responses` section.

```ts
const spec = new OpenApiSpecBuilder()
  .withOperation(
    'get', 
    '/', 
    new OperationSpecBuilder().withResponse(200, {
      content: {
        'text/json': {
          schema: {
            // put your schema here
          }
        }
    })
  )
  .build();
```

(Feel free to add new builder methods to further simplify your test code, just make sure to do that in a new commit and/or a new pull request.)

Reply:
Yes, I was shortcutting /w invalid spec based on knowledge of how the crawler operates.  Updated with OpenApiSpecBuilder but [initially patched in `components.schemas`](https://github.com/strongloop/loopback-next/pull/4365/files#diff-5f5f952350cff2452a82b82c113bca9dR57-R72).